plugins/flatpak/gs-flatpak.c

```diff
@@ -3719,32 +3723,22 @@ gs_flatpak_has_space_to_install (GsFlatpak *self, GsApp *app, GsAppList *list)
 }
 
 static gboolean
-gs_flatpak_has_space_to_update (GsFlatpak *self, GsApp *app, GsAppList *list)
+gs_flatpak_has_space_to_update (GsFlatpak *self, GsApp *app, GsAppList *list, gboolean is_auto_update)
 {
 	guint64 free_space = 0;
 	guint64 space_required = 0;
 
-	/* If an app has an updatable runtime, the runtime is not updated
-	 * as a part of app update.
-	 *
-	 * Having said that, runtime updates and their related refs (see T22716)
-	 * are done together and below heuristics might not be good for if GsApp
-	 * app is a runtime.
-	 */
-	if (gs_utils_app_is_auto_updating (app)) {
+	if (is_auto_update && gs_app_get_kind (app) != AS_APP_KIND_RUNTIME) {
```

Comment:
Notice that if the app needs a new runtime, the runtime's size will be included in the required space which is multipled. So if the filter for the runtime is to never require e.g. 1.6 GB * 2 + 500 MB, we need to do some extra checks here.

Reply:
I don't understand. If it's a **new** runtime and  we need to make it fit in that would require double disk space, no? (as it's a `--system` pull)

Reply:
Ah okay. I think you're right then.